query-service-impl/src/main/java/org/hypertrace/core/query/service/pinot/PinotBasedRequestHandler.java

```diff
@@ -463,7 +463,9 @@ private void measurePinotTagQueryExecutionTime(
   private void measureRequestAge(ExecutionContext executionContext) {
     try {
       Preconditions.checkNotNull(executionContext);
-      Preconditions.checkNotNull(executionContext.getQueryTimeRange());
+      if (executionContext.getTimeFilterColumn() == null) {
+        return;
```

Comment:
Don't think this is right. This is an invariant and a precondition check if the right way to handle this. If an invariant breaks, you want as much logs as possible. Because it's a condition that ALWAYS holds. If such a condition doesn't hold, it means something has gone wrong with your system.

Reply:
No, this condition is false many times. So logs were being cluttered by error stack trace. 
there are some pinot queries which doesn't have timeFilterColumn, for all of the logs will have an error stacktrace.
For example whenever a user will open full trace view page for trace, the error would be thrown.

Reply:
Move the check to the caller then. If this method is called, then `executionContext.getTimeFilterColumn()` is never NULL. 

Since it's a private method (whose arguments we control), you can remove the Precondtions check altogether. But do move this check to the caller.

Ref: https://stackoverflow.com/a/49952352

Reply:
done